hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -326,26 +326,41 @@ case class TableParallelize(rowsAndGlobal: IR, nPartitions: Option[Int] = None)
     val rowsAddr = ptype.loadField(res, 0)
     val nRows = rowsT.loadLength(rowsAddr)
 
-    val encRows = Array.tabulate(nRows) { i =>
-      if (rowsT.isElementMissing(rowsAddr, i))
-        fatal(s"cannot parallelize null values: found null value at index $i")
-      val baos = new ByteArrayOutputStream()
-      makeEnc(baos).writeRegionValue(ctx.r, rowsT.loadElement(rowsAddr, i))
-      baos.toByteArray
+    val nSplits = nPartitions.getOrElse(16)
+    val rowsPerPartition = nRows / nSplits
+    val leftOver = nRows - (nSplits * rowsPerPartition) // if partition index < leftOver, has rowsPerPartition + 1
```

Comment:
nSplits shouldn't be bigger than nRows.

You can use `utils.partition` to compute partition splits.  It's basically what you do here, except it has been debugged and has some nice checks.  The idiom we use in most places is:

```
        val partCounts = partition(n, nPartitions)
        val partStarts = partCounts.scanLeft(0)(_ + _)
```

Reply:
ah, I see. this is still correct, but can result in 0-element partitions.

Reply:
will switch.